docs/howto/build-guides/select-platforms.rst

```diff
@@ -0,0 +1,196 @@
+
+In this sample, building on AMD64 or 64-bit RISC-V will produce one charm that
+runs on Ubuntu 24.04 LTS across all architectures.
+
+.. important::
+
+    Charmcraft does not check that the resulting charm is architecture-independent.
+    It is up to the charm developer to ensure that the charm does not include any
+    architecture-dependent code, including Python dependencies that contain
+    compiled code.
+
+
+Select multiple build and target architectures
+----------------------------------------------
+
+Charms may contain architecture-specific code and thus need separate artifacts
+for each. You can declare multiple architectures in the ``platforms`` key,
+which instructs Charmcraft to build a charm for each of them as a set. For
+example, you could use the following snippet in ``charmcraft.yaml`` to build
+for both AMD64 and 64-bit RISC-V:
+
+.. code-block:: yaml
+
+   base: ubuntu@24.04
+   platforms:
+     amd64:
+       build-on: [amd64]
+       build-for: [amd64]
+     riscv64:
+       build-on: [riscv64]
+       build-for: [riscv64]
+
+Because the ``build-on`` and ``build-for`` entries are identical for each
+platform, the shorthand notation can be used instead:
+
+.. code-block:: yaml
+
+   base: ubuntu@24.04
+   platforms:
+     amd64:
+     riscv64:
+
+The results are the same with either snippet. Building on amd64 will
+produce one charm that runs on amd64. Building on ``riscv64`` will produce
+one charm that runs on ``riscv64``.
+
+.. note::
+
+    The ``build-for`` key may only contain one architecture, despite being a list.
+
+
+Select different architectures for building and running
+-------------------------------------------------------
+
+A charm may require cross-compilation to build. To create a charm for a
+different architecture, use the following ``charmcraft.yaml`` snippet:
+
+.. code-block:: yaml
+
+   base: ubuntu@24.04
+   platforms:
+     riscv64-cross:
+       build-on: [amd64]
+       build-for: [riscv64]
+
+Building on amd64 will produce one charm that runs on ``riscv64``.
+
+Note that the charm developer must ensure the charm is compatible with the
+target architectures. By default, the `charm`_,
+:ref:`python <craft_parts_python_plugin>`, :ref:`poetry <craft_parts_poetry_plugin>`,
+and :ref:`uv <craft_parts_uv_plugin>` plugins will install wheels for python packages
+for the ``build-on`` architecture rather than the ``build-for``. For more information,
+see `craft-parts#974`_.
+
+Select multiple OS releases
+---------------------------
+
+The resulting ``.charm`` file packed by charmcraft can only run on a single OS release
+or ``base``, using ``charmcraft.yaml`` nomenclature. A ``charmcraft.yaml`` can use
+multi-base syntax to create a set of charms, each for a different base. To do this, the
+base is defined in each platform entry instead of being defined with the
+top-level ``base`` and ``build-base`` keys.
+
+To build a charm for Ubuntu 22.04 and a charm for Ubuntu 24.04, use the
```

Comment:
```suggestion
To build a charm for Ubuntu 22.04 LTS and a charm for Ubuntu 24.04 LTS, use the
```

In the last paragraph, too.

Reply:
IMO this language is cumbersome. Not your fault I know... I just want to complain